### PR TITLE
2749 Fix ThreadDataLoader reset

### LINKED ISSUE #2749: ThreadDataLoader can't continuously run
**Describe the bug**
Currently, the `ThreadDataLoader` can only run for 1 epoch, it doesn't contain the reset logic.

Reproduce the issue:
```py
dataset = Dataset(data=self.datalist, transform=self.transform)
dataloader = ThreadDataLoader(dataset=dataset, batch_size=2, num_workers=0)

for d in dataloader:
    self.assertEqual(d["image"][0], "spleen_19.nii.gz")
    self.assertEqual(d["image"][1], "spleen_31.nii.gz")
    self.assertEqual(d["label"][0], "spleen_label_19.nii.gz")
    self.assertEqual(d["label"][1], "spleen_label_31.nii.gz")

for d in dataloader:
    print("will not print here.")
```

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2749 .

### Description
This PR fixed the continuous run issue of `ThreadDataLoader`.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Could you please help review this PR?

Thanks in advance.

**Review by wyli — COMMENTED** [review]
looks good to me, but @ericspod please confirm the usage. (the code comment line 91 is out of date in the current PR)

File: monai/data/thread_buffer.py [hunk 1]
```diff
@@ -87,8 +87,7 @@ class ThreadDataLoader(DataLoader):
     def __init__(self, dataset: Dataset, num_workers: int = 0, **kwargs):
         super().__init__(dataset, num_workers, **kwargs)
 
-        # ThreadBuffer will use the inherited __iter__ instead of the one defined below
-        self.buffer = ThreadBuffer(super().__iter__())
-
     def __iter__(self):
-        yield from self.buffer
+        # ThreadBuffer will use the inherited __iter__ instead of the one defined below
```
Comment (wyli) [inline monai/data/thread_buffer.py:91]: This comment is not valid any more. Could you help revise?

**Review by wyli — APPROVED** [review]

File: monai/data/thread_buffer.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/data/thread_buffer.py]: Sure, updated in the latest commit.
Thanks.

**Pull request merged**